### PR TITLE
Cart api

### PR DESCRIPTION
**Pull request by VedanthB**
- cart api services
- get cart items
- post cart item
- update cart item (qty)
- delete cart item

### CODE REVIEW

File: src/components/auth-pages/LoginCard.js [hunk 1]
```diff
@@ -1,84 +1,130 @@
...
+                        type="text"
+                        id="email"
+                        name="email"
+                        className="input-outlined input-outlined-cyan"
+                        placeholder=" "
+                        value={userData.email}
+                        onChange={handleChange}
+                    />
+                    <label
+                        style={{ backgroundColor: '#0000227c' }}
```
Comment (Ananyamadhu08) [inline src/components/auth-pages/LoginCard.js:44]: you can define a root variable to define colors that you are using repeatedly.

**Review by Ananyamadhu08 — APPROVED** [review]
Good job Vedanth! Your code is readable and clean. The things I liked is good variable naming    and folder structure. You can go ahead and merge!

File: src/App.js [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
+                hideProgressBar={false}
+                newestOnTop
+                closeOnClick
+                rtl={false}
+                pauseOnFocusLoss
+                draggable
+                pauseOnHover
+            />
+
+            {/* header / nav */}
```
Comment (deepak29-git) [inline src/App.js:28]: remove commented code if it is not used

File: src/components/auth-pages/SignupCard.js [hunk 3]
```diff
@@ -1,88 +1,185 @@
...
+    const navigate = useNavigate()
+
+    const handleChange = (e) => {
+        const { name, value } = e.target
+        setUserData({ ...userData, [name]: value })
+    }
+
+    return (
+        <div
+            style={{ margin: 'auto', backgroundColor: '#0000227c' }}
```
Comment (deepak29-git) [inline src/components/auth-pages/SignupCard.js:27]: avoid using inline style until and unless it is not needed

File: src/context/providers/CartProvider.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    const { encodedToken } = authState
+
+    useEffect(() => {
+        if (encodedToken) {
+            getCartItems(encodedToken, cartDispatch)
+        }
+    }, [encodedToken])
+
+    console.log(cartState)
```
Comment (deepak29-git) [inline src/context/providers/CartProvider.js:27]: remove console

File: src/utils/updateCartItem.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+        const {
+            data: { cart },
+            status,
+        } = await decrementCartItemService(encodedToken, itemId)
+
+        if (status >= 200 && status < 300) {
+            cartDispatch({ type: cartActions.GET_CART_ITEMS, payload: cart })
+
+            console.log(status, cart)
```
Comment (deepak29-git) [inline src/utils/updateCartItem.js:38]: remove console

**Review by deepak29-git — APPROVED** [review]

**Pull request merged**